### PR TITLE
Fix game spec

### PR DESCRIPTION
**Pull request by addod19**
git flow implemented....

### CODE REVIEW

**Comment by addod19** [comment]
git flow implemented

File: spec/Gemfile
Comment (bolah2009) [inline spec/Gemfile]: Let's keep this Gemfile at the root directory, not `spec/` directory. 

Also, is this line valid? What is it for? :thinking:

**Review by bolah2009 — CHANGES_REQUESTED** [review]
# Nice work team :clap: :clap:

Please this repo those not have Stickler linter activated, have you **on** it [here](https://stickler-ci.com)? :thinking: 
There's about `111 offenses detected` by `rubocop`, please let's fix this and submit another CRR.

#### :recycle:  Make these changes and submit another code review request. :+1: 

#### Feel free to reach me on slack if you having issues implementing these features. :+1: 

Good luck with your next submission. :star2: 

Cheers
[Bola](https://bolabuari.com/)

###### PS: Note that comments tagged as `OPTIONAL` are not part of the minimum requirements to approve this PR

**Review by maelfosso — CHANGES_REQUESTED** [review]
Hello @addod19 @maya88en 

Thanks for the improvements :smile: 

There are still some bug to fix.
Here is what I'm getting when trying to test it
```ruby
$ rspec
..F.................

Failures:

  1) Board display game board #display should display a grid
     Failure/Error: expect(tie.display).to eq display_test
     
       expected: "\n          |          |\n          X | X | O\n          -----------\n          |          |\n          O | X | X\n          -----------\n          |          |\n          X | 0 | O\n          "
            got: "\n    |          |\n    X | X | O\n    -----------\n    |          |\n    O | X | X\n    -----------\n    |          |\n    X | 0 | O\n    "
     
       (compared using ==)
     
       Diff:
       @@ -1,11 +1,11 @@
        
       -          |          |
       -          X | X | O
       -          -----------
       -          |          |
       -          O | X | X
       -          -----------
       -          |          |
       -          X | 0 | O
       -          
       +    |          |
       +    X | X | O
       +    -----------
       +    |          |
       +    O | X | X
       +    -----------
       +    |          |
       +    X | 0 | O
       +    
       
     # ./spec/board_spec.rb:55:in `block (4 levels) in <top (required)>'

Finished in 0.0386 seconds (files took 0.09772 seconds to load)
20 examples, 1 failure

Failed examples:

rspec ./spec/board_spec.rb:34 # Board display game board #display should display a grid
```

- [ x] RSpec on **Player** 

- [ ] RSpec on **Board**. The errors above comes from that file. 
There is a lot of method from the **Board** class that is not tested like 
- `place_marker`
- `valid_selection?`
- `winning_lines`
- `win?` and `tie?`
- Test `display` like checking if two strings are equals.

- [ x] RSpec on **Game**

Please kindly make the suggested change and submit a new code review request.

File: lib/board.rb
Comment (samgaco) [inline lib/board.rb:40]: - [ ] winning_lines should have a test too

File: lib/board.rb
Comment (samgaco) [inline lib/board.rb:48]: - [ ] they should be tested too, you can use mocks or [doubles](https://relishapp.com/rspec/rspec-mocks/docs/basics/test-doubles) to represent some of the variables inside those methods

**Review by samgaco — CHANGES_REQUESTED** [review]
Hi @addod19 and @maya88en :high_brightness: 

Some methods still to be tested before getting the approval, see the comments below.

Good spirit :hibiscus: 

------------------
Moving on :arrow_down: 

* **[For your next code review requests]**  Add a short summary of changes in Pull Request description

----------------
* ###### STATUS OF THE PROJECT: REQUESTED CHANGES :stop_sign:

**Review by bolah2009 — APPROVED** [review]
## Nice job implementing all suggested changes.

### It's a :+1: from me,

You can merge to master.

**Comment by addod19** [comment]
Testing TicTacToe with Ruby RSpec framework

**Pull request merged**
